worlds/sc2/MissionOrders.py

```diff
@@ -0,0 +1,389 @@
+            FillMission(MissionPools.MEDIUM, [MissionConnection(1, SC2Campaign.LOTV)], "Shakuras", completion_critical=True),
+            FillMission(MissionPools.MEDIUM, [MissionConnection(2, SC2Campaign.LOTV), MissionConnection(3, SC2Campaign.LOTV)], "Purifier", completion_critical=True),
+            FillMission(MissionPools.HARD, [MissionConnection(6, SC2Campaign.LOTV)], "Purifier", completion_critical=True),
+            FillMission(MissionPools.HARD, [MissionConnection(4, SC2Campaign.LOTV)], "Ulnar", completion_critical=True),
+            FillMission(MissionPools.HARD, [MissionConnection(6, SC2Campaign.LOTV)], "Tal'darim", completion_critical=True),
+            FillMission(MissionPools.HARD, [MissionConnection(5, SC2Campaign.LOTV), MissionConnection(7, SC2Campaign.LOTV)], "Return to Aiur", completion_critical=True),
+            FillMission(MissionPools.FINAL, [MissionConnection(8, SC2Campaign.LOTV)], "Return to Aiur", completion_critical=True),
+        ],
+        SC2Campaign.EPILOGUE: [
+            FillMission(MissionPools.VERY_HARD, [MissionConnection(12, SC2Campaign.WOL), MissionConnection(12, SC2Campaign.HOTS), MissionConnection(9, SC2Campaign.LOTV)], "_1", completion_critical=True),
+            FillMission(MissionPools.FINAL, [MissionConnection(0, SC2Campaign.EPILOGUE)], "_2", completion_critical=True),
+        ],
+        SC2Campaign.NCO: [
+            FillMission(MissionPools.EASY, [MissionConnection(-1, SC2Campaign.NCO)], "_1", completion_critical=True),
+            FillMission(MissionPools.MEDIUM, [MissionConnection(0, SC2Campaign.NCO)], "_1", completion_critical=True),
+            FillMission(MissionPools.MEDIUM, [MissionConnection(1, SC2Campaign.NCO)], "_2", completion_critical=True),
+            FillMission(MissionPools.HARD, [MissionConnection(2, SC2Campaign.NCO)], "_3", completion_critical=True),
+            FillMission(MissionPools.FINAL, [MissionConnection(3, SC2Campaign.NCO)], "_3", completion_critical=True),
+        ]
+    }
+
+
+smooth_difficulty = [MissionPools.EASY,
+                     MissionPools.MEDIUM, MissionPools.MEDIUM,
+                     MissionPools.HARD, MissionPools.HARD,
+                     MissionPools.VERY_HARD]
+max_difficulty = len(smooth_difficulty) - 1
+
+
+def make_golden_path(world: World, num_missions: int) -> List[FillMission]:
+    chain_name_options = ['Mar Sara', 'Char', 'Kaldir', 'Zerus', 'Skygeirr Station',
+                   'Dominion Space', 'Korhal', 'Aiur', 'Shakuras', 'Ulnar']
+    world.random.shuffle(chain_name_options)
+
+    first_chain = chain_name_options.pop()
+    first_mission = FillMission(MissionPools.STARTER, [MissionConnection(-1, SC2Campaign.GLOBAL)], first_chain,
+                                     completion_critical=True)
+
+    class Campaign:
+        def __init__(self, first_mission: FillMission, missions_remaining: int):
+            self.mission_order = [first_mission]
+            self.counter = 0
```

Comment:
Recommend renaming this to `mission_counter` or similar, I had to read a good chunk of `add_mission()` before I was sure what it was counting.

Reply:
Refactored them.